localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker_factory.py

```diff
@@ -149,8 +157,23 @@ def get_esm_worker(self) -> EsmWorker:
                 processor=esm_processor,
             )
         else:
-            raise Exception(
-                f"Unsupported event source mapping source service {source_service}. Please upvote or create a feature request."
+            poller_holder = {"create_poller": None}
+            lambda_hooks.create_event_source_poller.run(
+                poller_holder, source_service, self.esm_config
+            )
+
+            create_poller: Optional[Callable[..., Type[Poller]]] = poller_holder.get(
```

Comment:
Here as well, is the return type really a callable returning the type of poller, or the instance of poller?

Reply:
The return type is indeed a callable which we can use to construct the poller.

I opted for this approach since I did not want to pass the `source_arn`, `source_client`, and  `esm_processor` to the downstream hook.

Also, since we are using a factory pattern here, I prefer to keep the creation of the pollers in the driver function instead of downstream.

Reply:
```suggestion
            create_poller: Optional[Callable[..., Poller]] = poller_holder.get(
```